### PR TITLE
fix: Do not collect on raw_find_by_prefix

### PR DESCRIPTION
**Pull request by douglaz**

### CODE REVIEW

File: fedimint-core/src/db/mod.rs [hunk 1]
```diff
@@ -569,13 +569,24 @@ impl<'a> ISingleUseDatabaseTransaction<'a> for CommittableIsolatedDatabaseTransa
     }
 
     async fn raw_find_by_prefix(&mut self, key_prefix: &[u8]) -> Result<PrefixStream<'_>> {
-        let mut isolated = IsolatedDatabaseTransaction::new(self.dbtx.as_mut(), Some(self.prefix));
-        let stream = isolated
-            .raw_find_by_prefix(key_prefix)
-            .await?
-            .collect::<Vec<_>>()
```
Comment (elsirion) [inline fedimint-core/src/db/mod.rs:576]: Wow, I didn't remember this at all :grimacing: great catch!

**Review by elsirion — APPROVED** [review]

**Comment by elsirion** [comment]
Just needs a `clippy` fix apparently, then it should be good to go, great find! :)

**Comment by m1sterc001guy** [comment]
+1 Would prefer https://github.com/fedimint/fedimint/pull/2202

**Comment by douglaz** [comment]
https://github.com/fedimint/fedimint/pull/2202 is the way

**Pull request closed without merging**